# Logistic Regresion

# Importing the Libraries/Dataset

In [2]:
import pandas as pd

In [6]:
data = pd .read_csv("C:/Users/User/Desktop/Machine learning Project-Python/Final Folder/Dataset/breast_cancer.csv")
data.head(10)

Sample code number  Clump Thickness  Uniformity of Cell Size  \
0             1000025                5                        1   
1             1002945                5                        4   
2             1015425                3                        1   
3             1016277                6                        8   
4             1017023                4                        1   
5             1017122                8                       10   
6             1018099                1                        1   
7             1018561                2                        1   
8             1033078                2                        1   
9             1033078                4                        2   

   Uniformity of Cell Shape  Marginal Adhesion  Single Epithelial Cell Size  \
0                         1                  1                            2   
1                         4                  5                            7   
2                         1                  1                            2   
3                         8                  1                            3   
4                         1                  3                            2   
5                        10                  8                            7   
6                         1                  1                            2   
7                         2                  1                            2   
8                         1                  1                            2   
9                         1                  1                            2   

   Bare Nuclei  Bland Chromatin  Normal Nucleoli  Mitoses  Class  
0            1                3                1        1      2  
1           10                3                2        1      2  
2            2                3                1        1      2  
3            4                3                7        1      2  
4            1                3                1        1      2  
5           10                9                7        1      4  
6           10                3                1        1      2  
7            1                3                1        1      2  
8            1                1                1        5      2  
9            1                2                1        1      2

#To create a matrix of features with independent variables (x) and dependent variable (y)

In [9]:
x = data.iloc[:,1:-1].values
y = data.iloc[:,-1].values

# Splitting the dataset into training set and test set

In [11]:
from sklearn.model_selection import train_test_split

In [12]:
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=0.2, random_state=0)

# Traning the Logistic Regression model on a training set

In [13]:
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression(random_state=0)
classifier.fit(x_train, y_train)

LogisticRegression(random_state=0)

# Predicting the test set result

In [14]:
y_pred = classifier.predict(x_test)

# Making  the Confusion Matrix

In [15]:
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, y_pred)
print(cm)

[[84  3]
 [ 3 47]]


In [ ]:
This implies that there are 84+47 correct prediction and 3+3 incorrect prediction.
131 observations were correctly predicted.

# Computing the Accuracy with K-fold Cross Validation

In [19]:
from sklearn.model_selection import cross_val_score
accuracies = cross_val_score(estimator = classifier, X = x_train, y = y_train, cv = 10)
print("Accuracy: {:.2f} %".format(accuracies.mean()*100))
print("Standard Deviation: {:.2f} %".format(accuracies.std()*100))

Accuracy: 96.70 %
Standard Deviation: 1.97 %


The result shows that out of 100 observations, almost 97 will be correct and the standard deviation of the accuracies is low.
1.97% means that the 10 accuracies in the list more or less fall between 96.7±1.97.
Hence, the range of the accuracies is [94.7, 98.7], which shows a better prediction.
Hence, the logisti regression model performed well for the breast cancer prediction..